didcomm/interfaces/secrets_resolver.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from didcomm.types.types import JWK, KID, DID
+
+
+class SecretsResolver(ABC):
+    """Secrets resolver.
+
+    Retrieves JWKs with private keys for DIDs of key IDs.
+    """
+
+    @abstractmethod
+    async def get_key(self, kid: KID) -> JWK:
+        """Gets the JWK with the private key by the specified key ID."""
+        pass
+
+    @abstractmethod
+    async def get_keys(self, did: DID) -> List[JWK]:
+        """Gets all the JWKs with private keys for the specified DID."""
+        pass
+
+    @abstractmethod
+    async def find_keys(self, kids: List[KID]) -> List[JWK]:
+        """Finds the JWKs with private keys by the specified key IDs."""
```

Comment:
I would describe it a bit different:
For every key ID the method finds the corresponding private key and returns it in JWK format. The size of the output list can be less than the size of the input keyID list if not all key IDs have a corresponding private key in this secrets resolver.